src/broker/__tests__/txnOffsetCommit.spec.js

```diff
@@ -0,0 +1,125 @@
+const Broker = require('../index')
+const COORDINATOR_TYPES = require('../../protocol/coordinatorTypes')
+const {
+  secureRandom,
+  createTopic,
+  createConnection,
+  newLogger,
+  retryProtocol,
+} = require('testHelpers')
+
+describe('Broker > TxnOffsetCommit', () => {
+  let seedBroker,
+    consumerBroker,
+    transactionBroker,
+    topicName,
+    consumerGroupId,
+    transactionalId,
+    producerId,
+    producerEpoch
+
+  async function findBrokerForGroupId(groupId, coordinatorType) {
+    const {
+      coordinator: { host, port },
+    } = await retryProtocol(
+      'GROUP_COORDINATOR_NOT_AVAILABLE',
+      async () =>
+        await seedBroker.findGroupCoordinator({
+          groupId,
+          coordinatorType,
+        })
+    )
+
+    return new Broker({
+      connection: createConnection({ host, port }),
+      logger: newLogger(),
+    })
+  }
+
+  beforeEach(async () => {
+    transactionalId = `transaction-id-${secureRandom()}`
+    consumerGroupId = `group-id-${secureRandom()}`
+    topicName = `test-topic-${secureRandom()}`
+
+    seedBroker = new Broker({
+      connection: createConnection(),
+      logger: newLogger(),
+    })
+
+    await seedBroker.connect()
+    await createTopic({ topic: topicName, partitions: 4 })
+
+    transactionBroker = await findBrokerForGroupId(transactionalId, COORDINATOR_TYPES.TRANSACTION)
+    await transactionBroker.connect()
+    const result = await transactionBroker.initProducerId({
+      transactionalId,
+      transactionTimeout: 30000,
+    })
+
+    producerId = result.producerId
+    producerEpoch = result.producerEpoch
+
+    await transactionBroker.addPartitionsToTxn({
```

Comment:
@tulios I've added the preceding steps in the protocol here. You can see below we still don't get a rejection for invalid values.

Reply:
Yeah, well that's unfortunate, but I guess we will have more data once we start the actual producer. If we discover something new, we can come back add more tests.

Reply:
Sounds good to me @tulios. If we ultimately conclude that this is indeed Kafka's behavior today it may be prudent to raise an issue with the Kafka team to identify whether it's the intended behavior or a bug.